src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -368,6 +368,11 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
                 }
             }
 
+            if (TryGetInstanceExtensionParameter(out ParameterSymbol extensionParameter))
+            {
+                EnterParameter(extensionParameter);
```

Comment:
Should we also `LeaveParameter(extensionParameter);` below like we do for `methodThisParameter`?